File: DailyNews/src/main/java/com/jw/dailyNews/fragment/FragmentShouye.java
package com.jw.dailyNews.fragment;

import android.os.Handler;
import android.support.v4.view.ViewPager;
import android.view.Gravity;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.FrameLayout;
import android.widget.ImageButton;
import android.widget.PopupWindow;

import com.jw.dailyNews.R;
import com.jw.dailyNews.adapter.TabPageIndicatorAdapter;
import com.jw.dailyNews.base.BaseFragment;
import com.jw.dailyNews.utils.ThemeUtils;
import com.jw.dailyNews.wiget.LoadingPage;
import com.viewpagerindicator.TabPageIndicator;

/**
 * 创建时间：2017/6/21
 * 更新时间 2017/10/30 15:52
 * 版本：
 * 作者：Mr.jin
 * 描述：主界面的首页界面，其中内嵌多个新闻fragment，默认页面为推荐新闻页面,NewsTabObject,
 *       其他页面类型均为NewsTabNormal
 */

public class FragmentShouye extends BaseFragment implements View.OnClickListener{

    private ImageButton add;
    private PopupWindow popupWindow;
    private View view;
    private FrameLayout fl;
    private int popWindowHeight;
    private int popWindowWidth;
    private Animation dropDownAnim;
    private Animation dropUpAnim;

    @Override
    public View createSuccessView() {
        View view = View.inflate(getActivity(), R.layout.fragment_shouye, null);
        fl = (FrameLayout) view.findViewById(R.id.fl);
        add= (ImageButton) view.findViewById(R.id.add);
        add.setOnClickListener(this);
        TabPageIndicator indicator = (TabPageIndicator) view.findViewById(R.id.indicator);
        TabPageIndicatorAdapter adapter = new TabPageIndicatorAdapter(getChildFragmentManager());
        ViewPager pager = (ViewPager) view.findViewById(R.id.pager);
        pager.setAdapter(adapter);
        indicator.setViewPager(pager);
        popWindowHeight = ThemeUtils.getStatusBarHeight(getActivity())
                - ThemeUtils.getStatusBarHeight(getActivity())-45-fl.getMeasuredHeight();
        popWindowWidth = ThemeUtils.getWindowWidth(getActivity());
        dropDownAnim = AnimationUtils.loadAnimation(getActivity(), R.anim.drop_down);
        dropUpAnim = AnimationUtils.loadAnimation(getActivity(), R.anim.drop_up);
        return view;
    }

    @Override
    protected LoadingPage.LoadResult load() {
        return LoadingPage.LoadResult.success;
    }

    @Override
    public void onClick(View v) {

        switch (v.getId()){
            case R.id.add:
                if(popupWindow==null) {
                    if(view==null)
                        view = View.inflate(getActivity(), R.layout.layout_add, null);
                    popupWindow = new PopupWindow(view, popWindowWidth, popWindowHeight);
                    view.startAnimation(dropDownAnim);
                    int[] location = new int[2];
                    add.getLocationOnScreen(location);
                    popupWindow.showAtLocation(add, Gravity.NO_GRAVITY, 0, location[1]+fl.getMeasuredHeight());
                }
                else {
                    view.startAnimation(dropUpAnim);
                    new Handler().postDelayed(new Runnable() {
                        @Override
                        public void run() {
                            popupWindow.dismiss();
                            popupWindow=null;
                        }
                    },500);
                }
                break;
        }
    }
}

File: dailyNews/src/main/java/com/jw/dailyNews/fragment/FragmentShouye.kt
package com.jw.dailyNews.fragment

import android.os.Handler
import android.support.design.widget.TabLayout
import android.support.v4.content.ContextCompat
import android.support.v4.view.ViewCompat
import android.support.v4.view.ViewPager
import android.view.Gravity
import android.view.View
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.widget.FrameLayout
import android.widget.ImageButton
import android.widget.PopupWindow
import com.jw.dailyNews.R
import com.jw.dailyNews.adapter.TabPageIndicatorAdapter
import com.jw.dailyNews.base.BaseFragment
import com.jw.dailyNews.utils.ThemeUtils
import com.jw.dailyNews.wiget.LoadingPage
import com.viewpagerindicator.TabPageIndicator
import kotlinx.android.synthetic.main.fragment_shouye.*

/**
 * 创建时间：2017/6/21
 * 更新时间 2017/10/30 15:52
 * 版本：
 * 作者：Mr.jin
 * 描述：主界面的首页界面，其中内嵌多个新闻fragment，默认页面为推荐新闻页面,NewsTabObject,
 * 其他页面类型均为NewsTabNormal
 */

class FragmentShouye : BaseFragment(), View.OnClickListener {

    private var popupWindow: PopupWindow? = null
    private var vv: View? = null
    private var popWindowHeight: Int = 0
    private var popWindowWidth: Int = 0
    private lateinit var dropDownAnim: Animation
    private lateinit var dropUpAnim: Animation

    override fun createSuccessView(): View {
        val view = View.inflate(activity, R.layout.fragment_shouye, null)
        val ibAdd = view.findViewById<ImageButton>(R.id.ibAdd) as ImageButton
        ibAdd.setOnClickListener(this)
        val adapter = TabPageIndicatorAdapter(childFragmentManager)
        val vpShouye = view.findViewById<ViewPager>(R.id.vpShouye)
        val tpiShouye = view.findViewById<TabLayout>(R.id.tpiShouye)
        val flShouye = view.findViewById<FrameLayout>(R.id.flShouye)
        vpShouye.adapter=adapter
        tpiShouye.tabMode = TabLayout.MODE_SCROLLABLE;
        tpiShouye.setTabTextColors(ContextCompat.getColor(activity!!, R.color.gray), ContextCompat.getColor(activity!!, R.color.red))
        tpiShouye.setSelectedTabIndicatorColor(ContextCompat.getColor(activity!!, R.color.white))
        ViewCompat.setElevation(tpiShouye, 10f)
        tpiShouye.setupWithViewPager(vpShouye)

        popWindowHeight = (ThemeUtils.getStatusBarHeight(activity)
                - ThemeUtils.getStatusBarHeight(activity) - 45 - flShouye.measuredHeight)
        popWindowWidth = ThemeUtils.getWindowWidth(activity)
        dropDownAnim = AnimationUtils.loadAnimation(activity, R.anim.drop_down)
        dropUpAnim = AnimationUtils.loadAnimation(activity, R.anim.drop_up)
        return view
    }

    override fun load(): LoadingPage.LoadResult {
        return LoadingPage.LoadResult.success
    }

    override fun onClick(v: View) {

        when (v.id) {
            R.id.ibAdd -> if (popupWindow == null) {
                if (vv == null)
                    vv = View.inflate(activity, R.layout.layout_add, null)
                popupWindow = PopupWindow(view, popWindowWidth, popWindowHeight)
                vv!!.startAnimation(dropDownAnim)
                val location = IntArray(2)
                ibAdd.getLocationOnScreen(location)
                popupWindow!!.showAtLocation(ibAdd, Gravity.NO_GRAVITY, 0, location[1] + flShouye.measuredHeight)
            } else {
                vv!!.startAnimation(dropUpAnim)
                Handler().postDelayed({
                    popupWindow!!.dismiss()
                    popupWindow = null
                }, 500)
            }
        }
    }
}
